gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_long_prompt_optimization.ipynb

```diff
@@ -0,0 +1,1153 @@
+        "  <img width=\"20px\" src=\"https://upload.wikimedia.org/wikipedia/commons/5/51/Facebook_f_logo_%282019%29.svg\" alt=\"Facebook logo\">\n",
+        "</a>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0ccc35a93b9f"
+      },
+      "source": [
+        "| | | |\n",
+        "|-|-|-|\n",
+        "| Author | [Ivan Nardini](https://github.com/inardini) |"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tvgnzT1CKxrO"
+      },
+      "source": [
+        "##  I. Overview\n",
+        "\n",
+        "When developing Generative AI (Gen AI) applications, prompt engineering poses challenges due to its time-consuming and error-prone nature. Significant effort is involved when crafting and inputting prompts to achieve successful task completion. With the frequent release of foundational models, you face the added burden of migrating working prompts from one model version to another.\n",
+        "\n",
+        "Vertex AI prompt optimizer alleviates these challenges by providing an intelligent prompt optimization tool. With this tool you can both translate and optimize system instructions in the prompts and best demonstrations (examples) for prompt templates, which lets you shape LLM responses from any source model to a target Google model.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4HKyj5KwYePX"
+      },
+      "source": [
+        "### Objective\n",
+        "\n",
+        "This notebook demonstrates how to leverage Vertex AI prompt optimizer to optimize a long prompt for a Gemini model. The goal is to use Vertex AI prompt optimizer to find a new long prompt template that generates better responses.\n",
+        "\n",
+        "This tutorial uses the following Google Cloud services and resources:\n",
+        "\n",
+        "- Generative AI on Vertex AI\n",
+        "- Vertex AI prompt optimizer\n",
+        "- Vertex AI Gen AI evaluation\n",
+        "- Vertex AI Custom job\n",
+        "\n",
+        "The steps performed include:\n",
+        "\n",
+        "1. Define the prompt template you want to optimize.\n",
+        "2. Prepare the prompt optimization dataset.\n",
+        "3. Set optimization mode and steps.\n",
+        "5. Run the automatic prompt optimization job.\n",
+        "6. Collect the best prompt template and eval metric.\n",
+        "7. Validate the best prompt template."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "08d289fa873f"
+      },
+      "source": [
+        "### Dataset\n",
+        "\n",
+        "The dataset is a synthetic question/answer pairs for random short/medium/long periods are here. Here you have an example of Q&A record:\n",
```

Comment:
Can you rewrite this sentence? I'm a little bit confused by it.

Reply:
done